### PR TITLE
Enable reading the ES configuration from file

### PR DESCRIPTION
**Pull request by minhtuev**
With this change, we will attempt to create ES client basing on:
1) Function params
2) If none, from env variables
3) If none, from config file

The current PR attempts to solve this problem for EntityResolver and QuestionAnswerer. This change should enable MindMeld application to specify ES configuration more easily.

### CODE REVIEW

File: mindmeld/components/entity_resolver.py [hunk 1]
```diff
@@ -55,7 +55,14 @@ class EntityResolver:
...
     def __init__(
-        self, app_path, resource_loader, entity_type, es_host=None, es_client=None
+        self,
+        app_path,
+        resource_loader,
+        entity_type,
+        es_host=None,
+        es_username=None,
+        es_password=None,
+        es_client=None,
```
Comment (mhuang2173) [inline mindmeld/components/entity_resolver.py]: just wonder is there any reason that we have `es_client` in constructor of Entity Resolver but not Question Answerer? Does the passed-in `es_client` take precedence?

File: mindmeld/components/entity_resolver.py [hunk 2]
```diff
@@ -64,16 +71,43 @@ def __init__(
             resource_loader (ResourceLoader): An object which can load resources for the resolver
             entity_type: The entity type associated with this entity resolver
             es_host (str): The Elasticsearch host server
+            es_username (str): The Elasticsearch username server
```
Comment (mhuang2173) [inline mindmeld/components/entity_resolver.py]: typo: remove `server`?

File: mindmeld/components/entity_resolver.py [hunk 3]
```diff
@@ -64,16 +71,43 @@ def __init__(
...
+            es_password (str): The Elasticsearch password server
```
Comment (mhuang2173) [inline mindmeld/components/entity_resolver.py]: typo: remove `server`?

**Comment by vijay120** [comment]
@minhtuev Could you address the feedback?

**Pull request closed without merging**